decentraland/unity-explorer
# refactor: world & plugins initialization overhaul

## What is this pr?

Many plugins require access to the world and the player entity, but due to the application's initialization order, these were not available, which prevented the correct injection of dependencies into various classes, and forced us to make workarounds to add the dependencies at later stages.
The solution changes the creation order, making the world and the player entity available in the early stages, allowing them to be injected into different functionalities. Additionally, complexities in the codebase were removed.

It all started by trying to fix #1794 , which lead into refactoring how input groups are handled in the mvc code. I made an abstraction but it required the world dependency...

## How to test the changes?



1. Launch the explorer
2. An overall test is required through all the functionalities
3. Check that the application starts as expected
4. Interact with the authentication screen
5. Interact with the chat
6. Interact with the explore menu: backpack, minimap, settings
7. Interact with the connection status panel
8. Go to any scene that has UI in it
9. Interact with the passport
10. Interact with the sidebar
11. Follow #1794 steps. You should be able to hide the UI while the wheel is opened. Interact with the wheel.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
